delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/resolvers/ResolverResolutionRejection.scala

```diff
@@ -0,0 +1,73 @@
+package ch.epfl.bluebrain.nexus.delta.sdk.model.resolvers
+
+import ch.epfl.bluebrain.nexus.delta.kernel.utils.ClassUtils
+import ch.epfl.bluebrain.nexus.delta.rdf.IriOrBNode.{BNode, Iri}
+import ch.epfl.bluebrain.nexus.delta.rdf.Vocabulary.contexts
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.context.JsonLdContext.keywords
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.encoder.JsonLdEncoder
+import ch.epfl.bluebrain.nexus.delta.sdk.model.projects.ProjectRef
+import ch.epfl.bluebrain.nexus.delta.sdk.model.{Label, ResourceType}
+import io.circe.syntax._
+import io.circe.{Encoder, JsonObject}
+
+/**
+  * Enumeration of Resolver resolution rejection types.
+  *
+  * @param reason a descriptive message as to why the rejection occurred
+  */
+sealed abstract class ResolverResolutionRejection(val reason: String) extends Product with Serializable
+
+object ResolverResolutionRejection {
+
+  /**
+    * Rejection returned when a subject intends to retrieve a resource using a resolver at a specific revision,
+    * but the provided revision does not exist.
+    *
+    * @param provided the provided revision
+    * @param current  the last known revision
+    */
+  final case class RevisionNotFound(provided: Long, current: Long)
```

Comment:
Unless we want a report of what the failed resolutions when resolving a resource, we don't need this because we can fail first but another resource can have this id on a different project and may have this revision no ?

Reply:
Imagine you have just one resolver in a project, the default in project resolver (which is the normal case).
In this case you try to fetch a resource using that resolver and 4 things can happen (the same as the current logic):
- The revision provided is wrong
- The tag provided is wrong
- The project does not exist
- The resource id does not exist

If you have multiple resolvers, you will still have to save the last error and issue it back to the client